pkg/controller/build/reconciler_seeding_test.go

```diff
@@ -0,0 +1,418 @@
+			}
+			if !tt.expectSeeding && shouldSeed {
+				t.Errorf("Expected seeding not to be triggered but it would be")
+			}
+			if tt.expectNormalProcessing && !shouldNormalProcess {
+				t.Errorf("Expected normal processing to be triggered but it would not be")
+			}
+			if !tt.expectNormalProcessing && shouldNormalProcess {
+				t.Errorf("Expected normal processing not to be triggered but it would be")
+			}
+
+			if shouldSeed {
+				if preBuiltImage == "" {
+					t.Error("Pre-built image annotation exists but value is empty")
+				}
+			}
+		})
+	}
+}
+
+func TestCreateSyntheticMachineOSBuild(t *testing.T) {
+	tests := []struct {
+		name          string
+		buildName     string
+		imageSpec     string
+		poolName      string
+		expectedErr   bool
+		errorContains string
+	}{
+		{
+			name:        "Successful synthetic MOSB creation",
+			buildName:   "rendered-worker-12345",
+			imageSpec:   "registry.example.com/test@sha256:1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef",
+			poolName:    "worker",
+			expectedErr: false,
+		},
+		{
+			name:        "Valid layered pool MOSB creation",
+			buildName:   "rendered-layered-67890",
+			imageSpec:   "quay.io/example/layered@sha256:abcdef1234567890abcdef1234567890abcdef1234567890abcdef1234567890",
+			poolName:    "layered",
+			expectedErr: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Create test objects
+			mosc := &mcfgv1.MachineOSConfig{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-config",
+				},
+				Spec: mcfgv1.MachineOSConfigSpec{
+					MachineConfigPool:     mcfgv1.MachineConfigPoolReference{Name: tt.poolName},
+					RenderedImagePushSpec: "registry.example.com/test:latest",
+				},
+			}
+
+			mcp := &mcfgv1.MachineConfigPool{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: tt.poolName,
+				},
+				Spec: mcfgv1.MachineConfigPoolSpec{
+					Configuration: mcfgv1.MachineConfigPoolStatusConfiguration{
+						ObjectReference: corev1.ObjectReference{
+							Name: "rendered-" + tt.poolName + "-config",
+						},
+					},
+				},
+				Status: mcfgv1.MachineConfigPoolStatus{
+					Configuration: mcfgv1.MachineConfigPoolStatusConfiguration{
+						ObjectReference: corev1.ObjectReference{
+							Name: "rendered-" + tt.poolName + "-config",
+						},
+					},
+				},
+			}
+
+			// Set up fake clients
+			mcfgClient := fakemcfgclientset.NewSimpleClientset()
+
+			// Create a minimal reconciler with mocked listers
+			reconciler := &buildReconciler{
+				mcfgclient: mcfgClient,
+				listers: &listers{
+					machineConfigPoolLister: &fakeMCPLister{mcp: mcp},
+				},
+			}
+
+			ctx := context.Background()
+			result, err := reconciler.createSyntheticMachineOSBuild(ctx, mosc, tt.buildName, tt.imageSpec)
+
+			if tt.expectedErr {
+				if err == nil {
+					t.Errorf("Expected error but got none")
+				}
+				if tt.errorContains != "" && err != nil {
+					if !strings.Contains(err.Error(), tt.errorContains) {
```

Comment:
question (non-blocking): It looks like none of the test-cases use this path. Was it just an oversight?